frontend-react/src/components/shared/Table/Table.tsx

```diff
@@ -0,0 +1,225 @@
+    let SortIcon = Icon.SortArrow;
+    const isActive = columnTitle === activeColumn;
+    if (isActive && sortOrder === FilterOptionsEnum.ASC) {
+        SortIcon = Icon.ArrowUpward;
+    } else if (isActive && sortOrder === FilterOptionsEnum.DESC) {
+        SortIcon = Icon.ArrowDownward;
+    }
+    const isCurrentlyActiveColumn = columnTitle === activeColumn;
+
+    const handleHeaderClick = () => {
+        if (!isCurrentlyActiveColumn) {
+            // Reset active column and sort order on new column click
+            setActiveColumn(columnTitle);
+            setSortOrder(FilterOptionsEnum.ASC);
+        } else if (sortOrder === FilterOptionsEnum.NONE) {
+            // Explicitly set the proceeding sort order
+            setSortOrder(FilterOptionsEnum.ASC);
+        } else if (sortOrder === FilterOptionsEnum.ASC) {
+            setSortOrder(FilterOptionsEnum.DESC);
+        } else if (sortOrder === FilterOptionsEnum.DESC) {
+            setSortOrder(FilterOptionsEnum.NONE);
+        }
+    };
+    return (
+        <th
+            role="button"
+            tabIndex={0}
+            className={classnames({
+                "active-col-header": isActive,
+            })}
+            onClick={handleHeaderClick}
+        >
+            <div className="column-header-title-container">
+                <p>{columnTitle}</p>
+                {<SortIcon />}
+            </div>
+        </th>
+    );
+};
+
+const SortableTable = ({
+    rowData,
+    columnHeaders,
+}: {
+    rowData: RowData[][];
+    columnHeaders: string[];
+}) => {
+    const [activeColumn, setActiveColumn] = useState("");
+    const [sortOrder, setSortOrder] = useState(FilterOptionsEnum.NONE);
+    const sortedData = useMemo(() => {
+        return sortOrder !== FilterOptionsEnum.NONE && activeColumn
+            ? rowData.sort((a, b): number => {
+                  let contentColA = convertStringToSortableRepresentation(
+                      a.find((item) => item.columnKey === activeColumn)?.content
+                  );
+                  let contentColB = convertStringToSortableRepresentation(
+                      b.find((item) => item.columnKey === activeColumn)?.content
+                  );
+
+                  if (sortOrder === FilterOptionsEnum.ASC) {
+                      return contentColA < contentColB ? 1 : -1;
+                  } else {
+                      return contentColA > contentColB ? 1 : -1;
+                  }
+              })
+            : rowData;
+    }, [activeColumn, rowData, sortOrder]);
+    return (
+        <>
+            <thead>
+                <tr>
+                    {columnHeaders.map((header) => {
+                        return (
+                            <SortableTableHeader
+                                columnTitle={header}
+                                activeColumn={activeColumn}
+                                sortOrder={sortOrder}
+                                setActiveColumn={setActiveColumn}
```

Comment:
These setters also need to account for custom callbacks in case of API sorting.

Reply:
Let's talk about this feature synchronously! My gut feeling is that this should be in a separate PR / when we replace one of the tables that require API sorting.